apps/gitness/src/pages-v2/project/project-labels-list-container.tsx

```diff
@@ -75,11 +100,34 @@ export const ProjectLabelsList = () => {
     {
       onSuccess: (_data, variables) => {
         setOpenAlertDeleteDialog(false)
-        deleteLabel(variables.key)
+        deleteStoreSpaceLabel(variables.key)
       }
     }
   )
 
+  // temporary solution to fetch all label values - should be reworked once BE changes are done
+  useEffect(() => {
+    async function fetchAllLabelValues(storeLabels: ILabelType[]) {
+      if (!space_ref || !storeLabels) return
+
+      const valuesByKey: Record<string, any> = {}
+
+      for (const label of storeLabels) {
+        try {
+          const response = await fetch(`/api/v1/spaces/${space_ref}/labels/${label.key}/values`)
```

Comment:
we can do a small optimisation: check the count of values first, and only call API if count > 0.